extensions/amp-experiment/1.0/amp-experiment.js

```diff
@@ -262,12 +262,32 @@ export class AmpExperiment extends AMP.BaseElement {
   applyMutations_(experimentName, variantObject) {
     const doc = this.getAmpDoc();
     return doc.whenReady().then(() => {
-      // Parse / Validate all of our mutations
-      const mutationOperations = variantObject['mutations'].map(mutation =>
-        parseMutation(mutation, this.win.document)
-      );
+      /**
+       * Do a two step process:
+       * 1. Parse / Validate all mutations, so we can
+       * Bail out early if any mutation has an error, or
+       * exceeds the mutation limit
+       * 2. Apply all of the parsed mutations.
+       */
+      let totalMutations = 0;
+      let mutationOperations = [];
+      variantObject['mutations'].forEach(mutation => {
+        const mutationRecord = parseMutation(mutation, this.win.document);
 
-      // Apply our mutations
+        totalMutations += mutationRecord.mutations;
+        if (totalMutations > MAX_MUTATIONS) {
```

Comment:
I didn't see where we remove the old `MAX_MUTATIONS` check, does that mean we check the mutation count twice now?

Reply:
Nope, it is removed, and placed here instead, I think it may just be a weird git diff thing 😢

Reply:
So I retract what I said, It's totally still in there. I'm so sorry 😂 I don't know what I was doing when I opened this PR 😂